### PR TITLE
[FLOC-1540] Remove container API endpoint

### PR DESCRIPTION
**Pull request by itamarst**
https://clusterhq.atlassian.net/browse/FLOC-1540

### CODE REVIEW

**Comment by dwgebler** [comment]
The previous branch, create container endpoint, that this branched off has been merged now so I've merged `master` in to this branch to make review a bit easier.

File: flocker/control/httpapi.py [hunk 1]
```diff
@@ -485,6 +490,49 @@ def saved(_):
...
+        restarted again.
+        """,
+        examples=[
+            u"remove a container",
+            u"remove a container with unknown name",
+        ]
+    )
+    @structured(
+        inputSchema={
+            '$ref': '/v1/endpoints.json#/definitions/configuration_container'},
```
Comment (dwgebler) [inline flocker/control/httpapi.py]: :question: Is this actually required for a `DELETE` request? The tests will pass with an empty `inputSchema`. Also this is a question more broadly relating to how our JSON schemas are generated from the YAML files; specifying the input schema here would in one sense suggest that there ought to be a request body with the endpoint's defined required fields such as the container's host and image. Obviously we don't supply those and the API calls succeed on a `DELETE` without a request body and the `name` specified in the request path (which is entirely the correct behaviour I'd expect), but this is the case even with this input schema present, in which case I'm left wondering what purpose it serves here. So this is more a question of curiosity as to how this works.

**Comment by dwgebler** [comment]
Thanks @itamarst the OS X build failure is expected, I've also left an inline question that would be nice to hear your thoughts on, but otherwise this looks good to merge.

File: flocker/control/test/test_httpapi.py [hunk 2]
```diff
@@ -345,6 +345,90 @@ def test_response(self):
...
+        d = self.assertResponseCode(
+            b"POST", b"/configuration/containers",
+            {
+                u"host": self.NODE_A, u"name": name,
+                u"image": u"postgres"
+            }, CREATED
+        )
+        d.addCallback(lambda _: self.assertResult(
+            b"DELETE",
+            b"/configuration/containers/{}".format(name.encode("ascii")), None,
```
Comment (exarkun) [inline flocker/control/test/test_httpapi.py]: FWIW:

```
Python 3.4.0 (default, Apr 11 2014, 13:05:11) 
[GCC 4.8.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> b"{}".format(b"hello")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'bytes' object has no attribute 'format'
>>> u"{}".format(b"hello")
"b'hello'"
>>> u"{}".format(u"hello").encode("ascii")
b'hello'
>>> 
```

File: flocker/control/test/test_httpapi.py [hunk 3]
```diff
@@ -345,6 +345,90 @@ def test_response(self):
...
+        return self.assertResult(
+            b"DELETE",
+            b"/configuration/containers/%s" % (
+                unknown_name.encode('ascii'),),
+            None, NOT_FOUND,
+            {u"description": u'Container not found.'})
+
+    def _delete_test(self, name):
+        """
+        Create and then delete a container, ensuring the expected response
```
Comment (exarkun) [inline flocker/control/test/test_httpapi.py]: What response is expected?

File: flocker/control/test/test_httpapi.py [hunk 4]
```diff
@@ -345,6 +345,90 @@ def test_response(self):
...
+        d.addCallback(lambda _: self.assertResult(
+            b"DELETE",
+            b"/configuration/containers/{}".format(name.encode("ascii")), None,
+            OK, None,
+        ))
+        return d
+
+    def test_delete(self):
+        """
+        The ``DELETE`` action removes the given container from the
```
Comment (exarkun) [inline flocker/control/test/test_httpapi.py]: minor point - the HTTP spec calls them methods, not actions.

**Pull request merged**